virtualmachine/aws/s3util.go

```diff
@@ -0,0 +1,33 @@
+package aws
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+func getS3Client(region string) (svc *s3.S3, err error) {
+	if isRegionEmpty(region) {
+		region = getRegionFromEnv()
+		if isRegionEmpty(region) {
+			err = fmt.Errorf("Empty region provided")
+			return svc, err
```

Comment:
[NTH] just return would work here and at other places in this function as we have named the arguments... you will have to remove ":=" from  sess, err := getSession(region) below